libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+    *  on list and list items
+    */
+    handleKeyDown(event: any): boolean {
+        event.stopImmediatePropagation();
+        if (this.keyManager) {
+            if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+                return false;
+            } else if (event.keyCode === ENTER || event.keyCode === SPACE) {
+                this.updateNavigation(event);
+                return false;
+            }
+        }
+    }
+
+    private initializeDS(ds: FdpListDataSource<any>): void {
+        this.dsItems = [];
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+            if (this._dsSubscription) {
+                this._dsSubscription.unsubscribe();
+                this._dsSubscription = null;
+            }
+        }
+        // Convert ListDataSource<T> | T[] as DataSource
+        this._dataSource = this.openDataStream(ds);
+    }
+
+    private toDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        if (isDataSource(ds)) {
+            return ds;
+        }
+        return undefined;
+    }
+
+    private openDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        const initDataSource = this.toDataStream(ds);
+        if (initDataSource === undefined) {
+            throw new Error(`[dataSource] source did not match an array, Observable, or DataSource`);
+        }
+        /**
+         * This is single point of data entry to the component. We dont want to set data on different
+         * places. If any new data comes in either you do a search and you want to pass initial data
+         * its here.
+         */
+        this._dsSubscription = initDataSource
+            .open()
+            .pipe(takeUntil(this._destroyed))
+            .subscribe((data) => {
+                this.dsItems = data || [];
+                this._cd.markForCheck();
+            });
+        // initial data fetch
+        initDataSource.match('*');
+        return initDataSource;
+    }
+
+    scrollHandler(): void {
+        if (!this.loading && this.loadOnScroll) {
+            this.getMoreData();
+        }
+    }
+
+    /**
+   *  Handles lazy loading data
+   * onscroll and on more button click
+   */
+    public getMoreData(): void {
+        this.loading = true;
+        of(this._loadNewItems())
+            .pipe(
+                tap(data => {
+                    if (data === null || data === undefined) {
+                        console.error('===Invalid Response recived===');
+                    }
+                }),
+                delay(this.delayTime)
+            ).subscribe(result => {
+                if (result !== null && result !== undefined) {
+                    for (let i = this.items.length, j = 0; j < result.length; ++i, ++j) {
+                        this.items[i] = result[j];
+                    }
+                }
+                this.loading = false;
+                this._changeDetectorRef.markForCheck();
```

Comment:
look for any kinds of data changes and state changes. and inform parent

Reply:
updated here and in all applicable places for list.componen.ts